src/modules/governance/proposal/VotersListContainer.tsx

```diff
@@ -0,0 +1,150 @@
+
+import { formatProposal } from '../utils/formatProposal';
+import { VotersList } from './VotersList';
+import { VotersListModal } from './VotersListModal';
+
+type VotersListProps = {
+  proposal: CustomProposalType;
+};
+
+export type GovernanceVoter = {
+  _id: string; //mongodb objectid
+  aaveBalance: number; //2.6209877046554566
+  aavePropositionDelegate: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  aaveVotingDelegate: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  address: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  ensAvatar: string | null; //null
+  ensName: string | null; //null
+  isSybilVerified: boolean;
+  lastUpdateTimestamp: number; //1601928545
+  proposalHistory: unknown[]; //[]
+  propositionPower: number; //2.6209877046554566
+  propositionWeight: number; //1.6381173154096605e-7
+  stkAaveBalance: number; //0
+  stkAavePropositionDelegate: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  stkAaveVotingDelegate: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  twitterAvatar: string | null;
+  twitterHandle: string | null;
+  twitterName: string | null;
+  vote: 0 | 1; // 0 for nay, 1 for yae
+  votingHistory: unknown[]; //(2) [{…}, {…}]
+  votingPower: number; // the amount of voting power the user has - 2.6209877046554566
+  votingWeight: number; // the % that a single user contributes to the total - 1.6381173154096605e-7
+};
+
+export type VotersData =
+  | { yaes: GovernanceVoter[]; nays: GovernanceVoter[]; combined: GovernanceVoter[] }
+  | undefined;
+
+const sortByVotingPower = (a: GovernanceVoter, b: GovernanceVoter) => {
+  return a.votingPower < b.votingPower ? 1 : a.votingPower > b.votingPower ? -1 : 0;
+};
+
+export const VotersListContainer = (props: VotersListProps): JSX.Element => {
+  const { proposal } = props;
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<boolean>(false);
+  const [voters, setVoters] = useState<VotersData>();
+  const [votersModalOpen, setVotersModalOpen] = useState(false);
+
+  const votersUrl = `${process.env.NEXT_PUBLIC_API_BASEURL}/data/proposal-top-voters`;
+  const queryParams = `?proposal=${proposal.id}`;
+
+  useEffect(() => {
+    const getVoterInfo = async () => {
+      if (error) setError(false);
+
+      try {
+        // Get proposal voters data
+        const resp = await fetch(votersUrl + queryParams);
+
+        if (resp.ok) {
+          const [yaes, nays] = await resp.json();
+          // Transform data for UI, sort by highest voting power
+          const yesVoters = yaes.map((v: GovernanceVoter) => ({ ...v, vote: 1 }));
+          const noVoters = nays.map((v: GovernanceVoter) => ({ ...v, vote: 0 }));
+          const votersData = {
+            yaes: yesVoters.sort(sortByVotingPower),
+            nays: noVoters.sort(sortByVotingPower),
+            combined: yesVoters.concat(noVoters).sort(sortByVotingPower),
+          };
+          setVoters(votersData);
```

Comment:
I noticed there were some `any` typings here, would it help to create one type for the server response, and one for the UI data. Then you could type `yeas` and `naes` as the server response type, and then do something like
```
const yesVoters = yaes.map<GovernanceVoter>(v => ({ ...v, vote: 1 }));
const noVoters = nays.map<GovernanceVoter>(v => ({ ...v, vote: 0 }));
```

Reply:
There was one case where I put in `unknown[]` for `proposalHistory`.  And line 42 has the definition for `VotersData` where the `yaes` and `nays` are of type `GovernanceVoter[]`, but I failed to add in the types for the transformation statements here.  Great catch!  I'm pushing up the fixes for it.

Reply:
See https://github.com/aave/interface/pull/1244/commits/a88ebf6b858f14b37b47ebb9e8c68b1f81a3ac53